File: tests/fixtures/sample.java
package com.app.billing;

import com.app.core.Logger;

public interface PaymentProcessor {
    TransactionId process(long amount);
}

// Implements PaymentProcessor over the StripeClient SDK.
public class StripePaymentProcessor implements PaymentProcessor {
    private final String apiKey;
    private final Logger logger;

    public StripePaymentProcessor(String apiKey, Logger logger) {
        this.apiKey = apiKey;
        this.logger = logger;
    }

    @Override
    public TransactionId process(long amount) {
        logger.info("charging via StripeClient");
        return this.charge(amount);
    }

    private TransactionId charge(long amount) {
        return TransactionId.of(amount);
    }
}

File: tests/fixtures/sample.cs
using System;

namespace App.Billing
{
    public interface IPaymentProcessor
    {
        TransactionId Process(long amount);
    }

    // Implements IPaymentProcessor against the StripeClient SDK.
    public class StripePaymentProcessor : IPaymentProcessor
    {
        private readonly string _apiKey;

        public StripePaymentProcessor(string apiKey)
        {
            _apiKey = apiKey;
        }

        public TransactionId Process(long amount)
        {
            Console.WriteLine("charging via StripeClient");
            return Charge(amount);
        }

        private TransactionId Charge(long amount) => TransactionId.Of(amount);
    }
}
